### PR TITLE
Add Xamarin.Forms for Tizen support

### PR DESCRIPTION
**Pull request by vincent-kwon**
This PR enables Xamarin.Forms for Tizen support for SkiaSharp.

### CODE REVIEW

File: nuget/SkiaSharp.Views.Forms.nuspec [hunk 1]
```diff
@@ -67,7 +72,7 @@ Please visit https://go.microsoft.com/fwlink/?linkid=868517 to view the release
     <file platform="macos" src="lib/Xamarin.Mac20/SkiaSharp.Views.Forms.xml" />
     <file platform="windows" src="lib/uap10.0/SkiaSharp.Views.Forms.dll" />
     <file platform="windows" src="lib/uap10.0/SkiaSharp.Views.Forms.xml" />
-
+    <file platform="linux" src="lib/tizen40/SkiaSharp.Views.Forms.dll" />
```
Comment (rookiejava) [inline nuget/SkiaSharp.Views.Forms.nuspec]: doesn't add *.xml for tizen?

File: source/SkiaSharp.Views.Forms/SkiaSharp.Views.Forms.Native.Shared/SKGLViewRendererBase.cs [hunk 2]
```diff
@@ -124,6 +133,12 @@ protected override TNativeView CreateNativeControl()
 		{
 			return (TNativeView)Activator.CreateInstance(typeof(TNativeView), new[] { Context });
 		}
+#elif TIZEN4_0
+		protected virtual TNativeView CreateNativeControl()
+		{
+			TNativeView ret = (TNativeView)Activator.CreateInstance(typeof(TNativeView), new[] { TForms.Context.MainWindow });
```
[the remark this answers is not in the document]
Comment (rookiejava) [reply source/SkiaSharp.Views.Forms/SkiaSharp.Views.Forms.Native.Shared/SKGLViewRendererBase.cs]: return (TNativeView)Activator.CreateInstance(typeof(TNativeView), new[] { TForms.NativeParent });

File: samples/Basic/Xamarin.Forms/SkiaSharpSample.Tizen/SkiaSharpSample.Tizen.csproj [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </None>
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Tizen.NET" Version="4.0.0">
+      <ExcludeAssets>Runtime</ExcludeAssets>
+    </PackageReference>
+    <PackageReference Include="Tizen.NET.Sdk" Version="1.0.1" />
+    <PackageReference Include="Xamarin.Forms.Platform.Tizen" Version="2.5.0.77107" />
```
[the remark this answers is not in the document]
Comment (mattleibow) [reply samples/Basic/Xamarin.Forms/SkiaSharpSample.Tizen/SkiaSharpSample.Tizen.csproj]: Thanks for pointing this out. I usually use the lowest build/revision number for building/compatibility, but for Tizen we can bump it to the version we need.

File: samples/Gallery/Xamarin.Forms/Tizen/SkiaSharpGallery.Tizen.csproj [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      <Link>tpkroot\bin\runtimes\tizen-armel\native\libHarfBuzzSharp.so</Link>
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </None>
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Tizen.NET" Version="4.0.0">
+      <ExcludeAssets>Runtime</ExcludeAssets>
+    </PackageReference>
+    <PackageReference Include="Tizen.NET.Sdk" Version="1.0.1" />
```
[the remark this answers is not in the document]
Comment (vincent-kwon) [reply samples/Gallery/Xamarin.Forms/Tizen/SkiaSharpGallery.Tizen.csproj]: Checked but no change (it is the latest).

File: samples/Gallery/Xamarin.Forms/Tizen/SkiaSharpGallery.Tizen.csproj [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    <PackageReference Include="Xamarin.Forms.Platform.Tizen" Version="2.5.0.77107" />
```
[the remark this answers is not in the document]
Comment (vincent-kwon) [reply samples/Gallery/Xamarin.Forms/Tizen/SkiaSharpGallery.Tizen.csproj]: Done.

File: source/SkiaSharp.Views.Forms/SkiaSharp.Views.Forms.Native.Shared/SKCanvasViewRendererBase.cs [hunk 6]
```diff
@@ -118,6 +127,12 @@ protected override TNativeView CreateNativeControl()
 		{
 			return (TNativeView)Activator.CreateInstance(typeof(TNativeView), new[] { Context });
 		}
+#elif TIZEN4_0
+		protected virtual TNativeView CreateNativeControl()
+		{
+			TNativeView ret = (TNativeView)Activator.CreateInstance(typeof(TNativeView), new[] { TForms.Context.MainWindow });
```
[the remark this answers is not in the document]
Comment (vincent-kwon) [reply source/SkiaSharp.Views.Forms/SkiaSharp.Views.Forms.Native.Shared/SKCanvasViewRendererBase.cs]: Done.

File: source/SkiaSharp.Views.Forms/SkiaSharp.Views.Forms.Native.Shared/SKGLViewRendererBase.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (vincent-kwon) [reply source/SkiaSharp.Views.Forms/SkiaSharp.Views.Forms.Native.Shared/SKGLViewRendererBase.cs]: Done.

File: source/SkiaSharp.Views/SkiaSharp.Views.Tizen/ScalingInfo.cs [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+		{
+			return Elementary.GetProfile();
+		});
+
+		/// <summary>
+		/// DPI of the screen.
+		/// </summary>
+		private static Lazy<int> s_dpi = new Lazy<int>(() =>
+		{
+			Console.WriteLine("SEC--ScalingInfo  Profile : " + Profile);
```
[the remark this answers is not in the document]
Comment (vincent-kwon) [reply source/SkiaSharp.Views/SkiaSharp.Views.Tizen/ScalingInfo.cs]: Done.

File: source/SkiaSharp.Views/SkiaSharp.Views.Tizen/ScalingInfo.cs [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+			if (Profile == "tv")
+			{
+				// TV has fixed DPI value (72)
+				return 72;
+			}
+
+			int dpi = 0;
+			TSystemInfo.TryGetValue("http://tizen.org/feature/screen.dpi", out dpi);
+			Console.WriteLine("SEC--ScalingInfo  DPI : " + dpi);
```
[the remark this answers is not in the document]
Comment (vincent-kwon) [reply source/SkiaSharp.Views/SkiaSharp.Views.Tizen/ScalingInfo.cs]: Removed, Done.

File: source/SkiaSharp.Views/SkiaSharp.Views.Tizen/CustomRenderingView.cs [hunk 9]
```diff
@@ -0,0 +1,280 @@
...
+		}
+
+		/// <summary>
+		/// Creates the EFL controls.
+		/// </summary>
+		/// <param name="parent">The parent object.</param>
+		/// <returns>Pointer to the newly created control.</returns>
+		protected sealed override IntPtr CreateHandle(EvasObject parent)
+		{
+			Console.WriteLine("SEC-CustomRenderingView.CreateHandle   parent:" + parent);
```
[the remark this answers is not in the document]
Comment (vincent-kwon) [reply source/SkiaSharp.Views/SkiaSharp.Views.Tizen/CustomRenderingView.cs]: Done.

File: source/SkiaSharp.Views/SkiaSharp.Views.Tizen/CustomRenderingView.cs [hunk 10]
```diff
@@ -0,0 +1,280 @@
...
+
+			EvasImage = Interop.Evas.Image.evas_object_image_filled_add(Interop.Evas.evas_object_evas_get(handle));
+			Interop.Evas.Image.evas_object_image_colorspace_set(EvasImage, Interop.Evas.Image.Colorspace.ARGB8888);
+			Interop.Evas.Image.evas_object_image_smooth_scale_set(EvasImage, true);
+			Interop.Evas.Image.evas_object_image_alpha_set(EvasImage, true);
+
+			Interop.Elementary.elm_object_part_content_set(handle, "elm.swallow.content", EvasImage);
+
+			CreateNativeResources(parent);
+			Console.WriteLine("SEC-CustomRenderingView.CreateHandle   handle:" + handle);
```
[the remark this answers is not in the document]
Comment (vincent-kwon) [reply source/SkiaSharp.Views/SkiaSharp.Views.Tizen/CustomRenderingView.cs]: Done.

File: nuget/SkiaSharp.Views.Forms.nuspec (hunk 1 above)
Comment (vincent-kwon) [reply nuget/SkiaSharp.Views.Forms.nuspec]: Done.

**Pull request merged**